google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -447,7 +447,18 @@ public FirestoreOptions getOptions() {
 
   @Override
   public void close() throws Exception {
-    firestoreClient.close();
+    shutdown();
```

Comment:
I'm surprised this doesn't still call `firestoreClient.close()`. It looks like in `GrpcFirestoreRpc` you factored things to keep the existing behavior for `close()`.

Reply:
I don't have an opinion here - I do think it makes sense to keep the existing semantics, especially since we are adding a couple of new methods. If I call `close()` here the newly added shutdown tests time out though. Let me know if you want me to go back to calling close(), in which case I will remove the tests.

Reply:
Yeah, I think keeping the existing semantic is probably beneficial.

For updating the test we may be able to do something like:

```java
    ExecutorService testExecutorService = Executors.newSingleThreadExecutor();
    final Future<?> f = testExecutorService.submit(new Runnable() {
      @Override
      public void run() {
        firestore.close();
      }
    });
    
    try {
      f.get(5, TimeUnit.SECONDS);
      fail();
    } catch (TimeoutException e) {
      // expected
    } finally {
      testExecutorService.shutdown();
    }

```

Reply:
Updated. Thanks for the reviews.